### PR TITLE
Populate README with channels from conda_build_config.yaml

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [x] Added a ``news`` entry

Closes https://github.com/conda-forge/conda-smithy/issues/897 and unblocks https://github.com/conda-forge/conda-forge.github.io/pull/1620

### CODE REVIEW

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -720,9 +723,7 @@ def _render_ci_provider(
                 permit_undefined_jinja=True,
                 finalize=False,
                 bypass_env_check=True,
-                channel_urls=forge_config.get("channels", {}).get(
-                    "sources", []
-                ),
+                channel_urls=channel_sources,
```
Comment (jaimergp) [inline conda_smithy/configure_feedstock.py:726]: I am not sure about this change. I would appreciate some more info if someone knows!

File: conda_smithy/configure_feedstock.py [hunk 2]
```diff
@@ -1584,11 +1585,23 @@ def render_README(jinja_env, forge_config, forge_dir, render_info=None):
...
     if os.path.exists(ci_support_path):
         for filename in os.listdir(ci_support_path):
             if filename.endswith(".yaml"):
                 variant_name, _ = os.path.splitext(filename)
                 variants.append(variant_name)
+                with open(os.path.join(ci_support_path, filename)) as fh:
+                    data = yaml.safe_load(fh)
+                    for channel in data.get("channel_targets", ()):
+                        # channel_targets are in the form of "channel_name label"
+                        channel_targets.append(channel.split()[0])
```
Comment (isuruf) [inline conda_smithy/configure_feedstock.py:1598]:
```suggestion
                        channel_targets.append(channel.split(" "))
```

**Comment by xhochy** [comment]
@jaimergp Can you add a news entry, please?

**Comment by jaimergp** [comment]
Done! Thanks @xhochy

**Pull request merged**